src/vcpkg/base/files.cpp

```diff
@@ -2133,6 +2170,7 @@ namespace vcpkg
                     }
 
                     const auto full = base / entry->d_name;
+                    PushPopPath pushpop_outbase(out_base, entry->d_name);
```

Comment:
As far as I can see `outbase` is always equal to `full` so I'm not sure what this is doing?

Reply:
the initial base parameter is initialized differently depending on the initial call. Proximate initializes it to empty, which makes it only contain the subpath, whereas the normal functions initialize it with the original directory.